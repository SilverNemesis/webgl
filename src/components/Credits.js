import React from 'react';

const Credits = ({ show, credits }) => {
  if (!show) {
    return null;
  }
  let children
  if (credits) {
    children = credits.map((credit, index) => {
      if (credit.startsWith('link:')) {
        const url = credit.slice(5);
        return (
          <div key={index} className="credit"><a href={url} target="_blank" rel="noopener noreferrer">{url}</a></div>
        );
      } else if (credit === '') {
        return (
          <div key={index} className="credit"></div>
        );
      } else {
        return (
          <div key={index} className="credit">{credit}</div>
        );
      }
    });
  }
  return (
    <div id="credits">
      {children}
    </div>
  );
}

export default Credits;
