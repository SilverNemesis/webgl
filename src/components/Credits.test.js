import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Credits from './Credits'

const renderer = new ShallowRenderer();

it('renders correctly', () => {
    const showCredits = true;
    renderer.render(<Credits show={showCredits} credits={['These are credits', 'Stuff from guy', 'More stuff from other guy', 'link:http://localhost:3000/']} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
});
