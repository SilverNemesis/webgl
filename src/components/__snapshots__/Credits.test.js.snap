// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="credits"
>
  <div
    className="credit"
  >
    These are credits
  </div>
  <div
    className="credit"
  >
    Stuff from guy
  </div>
  <div
    className="credit"
  >
    More stuff from other guy
  </div>
  <div
    className="credit"
  >
    <a
      href="http://localhost:3000/"
      rel="noopener noreferrer"
      target="_blank"
    >
      http://localhost:3000/
    </a>
  </div>
</div>
`;
